ratis-client/src/main/java/org/apache/ratis/client/impl/RaftClientImpl.java

```diff
@@ -17,6 +17,7 @@
  */
 package org.apache.ratis.client.impl;
 
+import jdk.nashorn.internal.ir.annotations.Immutable;
```

Comment:
@OneSizeFitsQuorum , Please use `javax.annotation.concurrent.Immutable`.

Reply:
It looks like this will fail to compile.
<img width="948" alt="image" src="https://github.com/apache/ratis/assets/32640567/179908a7-6427-456a-9878-bc335726be4a">
<img width="1512" alt="image" src="https://github.com/apache/ratis/assets/32640567/58aeeadd-952f-4088-8b4a-c9b95d9d12d7">

Reply:
@OneSizeFitsQuorum , sorry that I was not clear.  The tag is for the class as below:
```java
+++ b/ratis-common/src/main/java/org/apache/ratis/protocol/RaftPeerId.java
@@ -22,6 +22,7 @@ import org.apache.ratis.thirdparty.com.google.protobuf.ByteString;
 import org.apache.ratis.util.JavaUtils;
 import org.apache.ratis.util.Preconditions;
 
+import javax.annotation.concurrent.Immutable;
 import java.nio.charset.StandardCharsets;
 import java.util.Map;
 import java.util.Objects;
@@ -33,6 +34,7 @@ import java.util.function.Supplier;
  * <p>
  * This is a value-based class.
  */
+@Immutable
 public final class RaftPeerId {
   private static final Map<ByteString, RaftPeerId> BYTE_STRING_MAP = new ConcurrentHashMap<>();
   private static final Map<String, RaftPeerId> STRING_MAP = new ConcurrentHashMap<>();
```

Reply:
Sorry, thank you for your patient guidance ~ now it is no problem! I learned something new!